src/status_im/ui/components/large_toolbar/view.cljs

```diff
@@ -0,0 +1,85 @@
+  (:require-macros [status-im.utils.views :as views]))
+
+;; header-in-toolbar - component - small header in toolbar
+;; nav-item          - component/nil - if nav-item like back button is needed, else nil
+;; action-items      - status-im.ui.components.toolbar.view/actions 
+(defn minimized-toolbar [header-in-toolbar nav-item action-items anim-opacity]
+  (let [has-nav? (boolean nav-item)]
+    [toolbar/toolbar
+     {:transparent? true
+      :style        styles/minimized-toolbar}
+     nav-item
+     [react/animated-view
+      {:style (cond-> (styles/animated-content-wrapper anim-opacity)
+                (false? has-nav?)
+                (assoc :margin-left -40 :margin-right 40))}
+      header-in-toolbar]
+     action-items]))
+
+;; header   - component that serves as large header without any top/bottom padding
+;;            top(4px high) and bottom(16px high and with border) padding
+;;            are assumed to be constant
+;;            this is wrapped with padding components and merged with content
+;; content  - vector - of the rest(from header) of the list components
+;;            wrapped header and content form the data prop of flat-list
+;; list-ref - atom - a reference to flat-list for the purpose of invoking its
+;;            methods
+;; scroll-y - animated value tracking the y scoll of the main content in flat-list-view
+(views/defview flat-list-with-large-header [header content list-ref scroll-y]
+  (views/letsubs [window-width [:dimensions/window-width]]
+    (let [header-top-padding [react/view {:height 4}]
+          ;; header bottom padding with border-bottom
+          ;; fades out as it approaches toolbar shadow
+          header-bottom      [react/animated-view
+                              {:style (styles/flat-list-with-large-header-bottom scroll-y)}]
+          wrapped-data       (into [header-top-padding header header-bottom] content)]
+      [react/view {:flex 1}
+       ;; toolbar shadow
+       [react/animated-view
+        {:style (styles/flat-list-with-large-header-shadow window-width scroll-y)}]
+
+       [list.views/flat-list
+        {:style                          styles/flat-list
+         :data                           wrapped-data
+         :initial-num-to-render          3
+         :ref                            #(when % (reset! list-ref (.getNode %)))
+         :render-fn                      list.views/flat-list-generic-render-fn
+         :key-fn                         (fn [item idx] (str idx))
+         :scrollEventThrottle            16
+         :on-scroll                      (animation/event
+                                          [{:nativeEvent {:contentOffset {:y scroll-y}}}]
+                                          {:useNativeDriver true})
+         :keyboard-should-persist-taps   :handled}
+        {:animated? true}]])))
+
+(defn generate-view
+  [header-in-toolbar nav-item toolbar-action-items header content list-ref]
+  "main function which generates views.
```

Comment:
the docstring is before the arguments, if you put it after it is evaluated, because it is a string and there is code afterward it is not noticeable, but that is not considered a docstring and won't be picked up by dev tools for instance.

Reply:
ups:)